pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -494,6 +522,31 @@ func (qjm *XController) GetQueueJobsEligibleForPreemption() []*arbv1.AppWrapper
 
 	if !qjm.isDispatcher { // Agent Mode
 		for _, value := range queueJobs {
+
+			if value.Status.State == arbv1.AppWrapperStateActive && value.Spec.SchedSpec.DispatchDuration.Limit > 0 {
+				conditionsLen := len(value.Status.Conditions)
+				var condition arbv1.AppWrapperCondition
+				for i := (conditionsLen - 1); i > 0; i-- {
```

Comment:
We should move this logic outside of the conditionals for the `DispatchDuration` and the `minAvailable`. It is the same loop looking for the `Dispatch` condition, so why do it twice? I know conditions is not supposed to be a huge array so it is O(1) mostly and not very intense operation, but I think we should move it, either way. It looks better and it is more efficient.